coalib/bears/Bear.py

```diff
@@ -305,6 +314,36 @@ def log_message(self, log_message, timestamp=None, **kwargs):
     def run(self, *args, dependency_results=None, **kwargs):
         raise NotImplementedError
 
+    def setup_profile(self, profiler=None):
+        filename = '{}_{}.prof'.format(self.section.name, self.name)
+        if self.profile is True:
+            profiler.dump_stats(filename)
+        elif os.path.exists(self.profile) and os.path.isdir(self.profile):
+            profiler.dump_stats(os.path.join(self.profile, filename))
+        elif not os.path.isfile(self.profile):
+            os.mkdir(os.path.join(self.profile))
+            profiler.dump_stats(os.path.join(self.profile, filename))
```

Comment:
not sure this behaviour is too complicated... why not just allowing `True`/`False` and put it either into the working directory or a subdirectory like `profile`/`profile-data`/`bear-profiling`/... ?

Reply:
No its not, If I understand here correctly then you are talking about this `filename = '{}_{}.prof'.format(self.section.name, self.name)` this part was added in cEP and also it is also giving a useful information like suppose some bear `ABCBear` in 2 different sections of `.coafile` then it will create a files with name `sectionOne_ABCBear` and `sectionTwo_ABCBear`. (here section name is my assumption)

Reply:
>not sure this behaviour is too complicated

I don't think its complicated :sweat_smile:

Reply:
> you are talking about this filename = '{}_{}.prof'.format(self.section.name, self.name)

Nope, that is totally okay :) However, with the above code changes by you, my doubts may do not apply any more. So nvm for now :3